bots/telegram.js

```diff
@@ -58,7 +70,21 @@ internal.htmlEscape = str => (
     .replace(/>/g, '&gt;')
 );
 
-client.on('text', (ctx, next) => {
+const downloadLinkMiddleware = (ctx, next) => {
+  if (ctx.updateSubType !== 'text') {
+    const updateSubType = ctx.message[ctx.updateSubType];
+    return client.telegram.getFileLink(updateSubType.file_id)
+      .then(link => {
+        const filename = uuid() + path.extname(url.parse(link).pathname);
+        http(link).pipe(fs.createWriteStream(path.join(files.directory, filename)));
+        ctx.state.fileLink = new url.URL(filename, files.baseUrl);
+        return next();
+      });
+  }
+  return next();
```

Comment:
I'd prefer that to be optional. Let's disable the functionality if `files.directory` is not defined.